lib/Doctrine/ORM/EntityRepository.php

```diff
@@ -129,9 +131,13 @@ public function createNativeNamedQuery($queryName)
      * Clears the repository, causing all managed entities to become detached.
      *
      * @return void
+     *
+     * @deprecated 2.7 This method is being removed from the ORM and won't have any replacement
      */
     public function clear()
     {
+        @trigger_error('Method ' . __METHOD__ . '() is deprecated and will be removed in Doctrine ORM 3.0.', E_USER_DEPRECATED);
```

Comment:
This will create two deprecation messages since the called method would also trigger an error, not really a huge issue but might confuse people (I think). 

We could add an error handler and then restore it back to avoid it or could not add a message here (but that might be confusing for users). What do you think is the most appropriate one?

Reply:
for my case it would be better to remove the parameter, but you already say that its is a bc break for you.

to add an error handler, could create much more problems in custom use cases, as the double notification.

so i think its better to trigger 2 messages as possible problems with error handlers.